### PR TITLE
impl manage retweet

### PR DESCRIPTION
**Pull request by mmmommm**
Implement PostRetweet and UndoRetweet method.

### CODE REVIEW

**Comment by sivchari** [comment]
please add example

**Comment by sivchari** [comment]
Please update push target to dev-1.1. After that, I check your PR.

**Comment by mmmommm** [comment]
@sivchari 
Sorry my mistake.
I fixed it, please review.

File: client.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ type Tweets interface {
 	ConnectToStream(ctx context.Context, ch chan<- ConnectToStreamResponse, errCh chan<- error, opt ...*ConnectToStreamOption) *ConnectToStream
 	VolumeStreams(ctx context.Context, ch chan<- VolumeStreamsResponse, errCh chan<- error, opt ...*VolumeStreamsOption) *VolumeStreams
 	RetweetsLookup(ctx context.Context, tweetID string, opt ...*RetweetsLookupOption) (*RetweetsResponse, error)
+	PostRetweet(ctx context.Context, uid string, tid string) (*PostRetweetResponse, error)
+	UndoRetweet(ctx context.Context, id string, stid string) (*UndoRetweetResponse, error)
```
Comment (sivchari) [inline client.go:30]: I think some parameters (e.g. uid, tid) are not clearly.
This args are shown user. So, we must define arg name more meaningful.

File: retweet.go [hunk 2]
```diff
@@ -51,3 +52,86 @@ func retweetsLookup(ctx context.Context, c *client, tweetID string, opt ...*Retw
...
+
+// uid = "user_id" tid = "tweet_id"
+func postRetweet(ctx context.Context, c *client, uid string, tid string) (*PostRetweetResponse, error) {
+	if uid == "" {
+		return nil, errors.New("post retweet by uid: uid parameter is required")
+	}
+	ep := fmt.Sprintf(postRetweetURL, uid)
+
+	if tid == "" {
+		return nil, errors.New("post retweet by tid: tid parameter is required")
```
Comment (sivchari) [inline retweet.go:64]:
```suggestion
		return nil, errors.New("post retweet by tweet id: tweet id parameter is required")
```

File: retweet.go [hunk 3]
```diff
@@ -51,3 +52,86 @@ func retweetsLookup(ctx context.Context, c *client, tweetID string, opt ...*Retw
...
+	return &postRetweet, nil
+}
+
+// stid = "source_tweet_id"
+func undoRetweet(ctx context.Context, c *client, id string, stid string) (*UndoRetweetResponse, error) {
+	if id == "" {
+		return nil, errors.New("undo retweet by id: id parameter is required")
+	}
+	if stid == "" {
+		return nil, errors.New("undo retweet by source_tweet_id: source_tweet_id parameter is required")
```
Comment (sivchari) [inline retweet.go:108]:
```suggestion
		return nil, errors.New("undo retweet by sourceTweetId: sourceTweetId parameter is required")
```

**Review by sivchari — CHANGES_REQUESTED** [review]

File: client.go (hunk 1 above)
Comment (mmmommm) [reply client.go:37]: I have changed args name to
```
uid -> userID
tid -> tweetID
stid -> sourceTweetID
```

Comment (mmmommm) [reply client.go:37]: I'll change #112 as well.

File: retweet.go [hunk 4]
```diff
@@ -51,3 +52,84 @@ func retweetsLookup(ctx context.Context, c *client, tweetID string, opt ...*Retw
 
 	return &retweetsLookup, nil
 }
+
+func postRetweet(ctx context.Context, c *client, userID string, tweetID string) (*PostRetweetResponse, error) {
+	if userID == "" {
+		return nil, errors.New("post retweet by userID: userID parameter is required")
```
Comment (sivchari) [inline retweet.go:58]: I think this error message isn't good.
This is because some of the error messages are duplicated.

File: retweet_test.go [hunk 5]
```diff
@@ -220,3 +220,183 @@ func Test_retweetsLookup(t *testing.T) {
...
+			},
+			want: &gotwtr.UndoRetweetResponse{
+				Errors: []*gotwtr.APIResponseError{
+					{
+						Message: "Sorry, that page does not exist, code:34",
+					},
+				},
+			},
+			wantErr: true,
+		},
```
Comment (sivchari) [inline retweet_test.go:384]: Please add more test of invalid case (e.g. request parameter is invalid)

**Review by sivchari — CHANGES_REQUESTED** [review]

File: retweet.go (hunk 4 above)
Comment (mmmommm) [reply retweet.go:58]: sorry, I'll change to `post retweet: userID parameter is required`

File: error.go [hunk 6]
```diff
@@ -25,6 +25,7 @@ type APIResponseError struct {
 	RequiredEnrollment string      `json:"required_enrollment,omitempty"`
 	RegistrationURL    string      `json:"registration_url,omitempty"`
 	ConnectionIssue    string      `json:"connection_issue,omitempty"`
+	Status             int         `json:"status,omitempty"`
```
Comment (mmmommm) [inline error.go:28]: I added this field because we need it when we get a 403 error.

File: retweet_test.go [hunk 7]
```diff
@@ -220,3 +220,425 @@ func Test_retweetsLookup(t *testing.T) {
...
+		{
+			name: "403 authentication error",
```
Comment (mmmommm) [inline retweet_test.go:508]: Here is 403 error

File: retweet_test.go [hunk 8]
```diff
@@ -220,3 +220,425 @@ func Test_retweetsLookup(t *testing.T) {
...
+							{
+								"title": "Unsupported Authentication",
+								"detail": "Authenticating with OAuth 2.0 Application-Only is forbidden for this endpoint.  Supported authentication types are [OAuth 1.0a User Context, OAuth 2.0 User Context].",
+								"type": "https://api.twitter.com/2/problems/unsupported-authentication",
+								"status": 403
+							}
+						]
+					}`
+					return &http.Response{
+						StatusCode: http.StatusBadRequest,
```
Comment (sivchari) [inline retweet_test.go:312]: StatusBadRequest is 400.
This status code is right ?

Comment (mmmommm) [reply retweet_test.go]: Ah, sorry. StatusForbidden is right.

**Review by sivchari — APPROVED** [review]
LGTM

**Pull request merged**